### PR TITLE
PR: Update Codicons update command and assets

### LINKED ISSUE #252: Improve Codicons update command
Follow up to the elements point out at: https://github.com/spyder-ide/qtawesome/pull/241#issue-1799283041

The update command for Codicons (`update_msc`) pulls directly from the `vscode-codicons` repo `main` branch. That can leave the font after running the update command in a a state which doesn't correspond to an actual released version of the icon font. The command can be run actually to get a version update only when the last commit over the `vscode-codicons` repo is the one done when doing a release.

Probably we should refactor how the command works and instead of pulling from `main` use the latest releases URL and get the needed files from the release assets as is done with the FontAwesome update command

### PR DESCRIPTION
**Pull request by dalthviz**
## Changes description
* Change to use `.css` instead of `.csv` file to build charmap
* Add `--msc-version` arg to the command and use it to build the download URLs (.ttf, .css and .json files)
* Update codicon assets

Fixes #252

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ The following prefixes are currently available to use:
 
 - `ri` prefix holds [**Remix Icon** 2.5.0 with its 2271 icons.](https://github.com/Remix-Design/RemixIcon)
 
-- `msc` prefix holds Microsoft's [**Codicons** 0.0.35 with its 446 icons.](https://github.com/microsoft/vscode-codicons)
```
Comment (ccordoba12) [inline README.md:74]: How is that the version is the same but the number of icons is different? Shouldn't the version number be greater than `0.0.35`?

File: setupbase.py [hunk 2]
```diff
@@ -253,24 +250,25 @@ def run(self):
...
 
     def initialize_options(self):
-        """Required by setuptools."""
+        """Set default values for the command options."""
+        self.msc_version = ''
 
     def finalize_options(self):
-        """Required by setuptools."""
+        """Validate the command options."""
+        assert bool(self.msc_version), 'Codicons version is mandatory for this command.'
```
Comment (ccordoba12) [inline setupbase.py:271]:
```suggestion
        assert bool(
            self.msc_version
        ), 'Codicons version is mandatory for this command.'
```

Make this fit in 79 columns

File: setupbase.py [hunk 3]
```diff
@@ -279,29 +277,32 @@ def __print(self, msg):
     def run(self):
         """Run command."""
 
-        # Download .csv to a temporary path:
-        package_json = urlopen(self.DOWNLOAD_URL_JSON)
+        # Download .json to a temporary path:
+        download_url_json = self.DOWNLOAD_URL_JSON.format(version=self.msc_version)
```
Comment (ccordoba12) [inline setupbase.py:281]:
```suggestion
        download_url_json = self.DOWNLOAD_URL_JSON.format(
            version=self.msc_version
        )
```

File: setupbase.py [hunk 4]
```diff
@@ -279,29 +277,32 @@ def __print(self, msg):
...
+        package_json = urlopen(download_url_json)
         package_info = json.load(package_json)
         package_version = package_info['version']
+        assert self.msc_version == package_version, 'Codicons version does not match with `package.json` info. %s and %s' % (self.msc_version, package_version)
```
Comment (ccordoba12) [inline setupbase.py:285]:
```suggestion
        assert self.msc_version == package_version, (
            "Codicons version does not match with `package.json` info. %s and %s"
            % (
                self.msc_version,
                package_version,
            )
        )
```

File: setupbase.py [hunk 5]
```diff
@@ -279,29 +277,32 @@ def __print(self, msg):
...
         self.__print('Will download codicons version: %s' % package_version)
 
-        # Download .csv to a temporary path:
-        with tempfile.NamedTemporaryFile(mode='wb+', suffix='.csv', prefix='codicon', delete=False) as tempCSV:
-            self.__print('Downloading: %s' % self.DOWNLOAD_URL_CSV)
-            response = urlopen(self.DOWNLOAD_URL_CSV)
-            shutil.copyfileobj(response, tempCSV)
+        # Download .css:
+        donwload_url_css = self.DOWNLOAD_URL_CSS.format(version=self.msc_version)
```
Comment (ccordoba12) [inline setupbase.py:289]:
```suggestion
        donwload_url_css = self.DOWNLOAD_URL_CSS.format(
            version=self.msc_version
        )
```

File: setupbase.py [hunk 6]
```diff
@@ -310,12 +311,13 @@ def run(self):
             json.dump(charmap, f, indent=4, sort_keys=True)
 
         # Dump a .ttf font file:
+        download_url_ttf = self.DOWNLOAD_URL_TTF.format(version=self.msc_version)
```
Comment (ccordoba12) [inline setupbase.py:314]:
```suggestion
        download_url_ttf = self.DOWNLOAD_URL_TTF.format(
            version=self.msc_version
        )
```

**Review by ccordoba12 — COMMENTED** [review]
Some stylistic questions and important question for you @dalthviz.

File: README.md (hunk 1 above)
Comment (dalthviz) [reply README.md:74]: I think this is caused due to the change from the `.csv` to the `.css` file as source to generate the icons charmap file. As mentioned at https://github.com/spyder-ide/qtawesome/pull/241#issue-1799283041 the `.csv` misses icons definitions available from the `.css` file.

**Review by ccordoba12 — APPROVED** [review]
Looks good to me now, thanks @dalthviz!

**Pull request merged**